arraycontext/impl/pyopencl/taggable_cl_array.py

```diff
@@ -73,13 +73,7 @@ def copy(self, queue=cla._copy_queue, tags=None, axes=None, _new_class=None):
             ``self.__class__``.
         """
         _new_class = self.__class__ if _new_class is None else _new_class
-
-        if queue is not cla._copy_queue:
-            # Copying command queue is an involved operation, use super-class'
-            # implementation.
-            base_instance = super().copy(queue=queue)
-        else:
-            base_instance = self
+        base_instance = super().copy(queue=queue)
```

Comment:
@kaushikcfd I'm not sure what the intended effect was here, but it seems like in the else case it doesn't do a copy?

Reply:
Yep, that was the idea. Did this run into any issues?

Reply:
Yeah, when trying to do a proper copy of a `TaggableCLArray` with `ary.copy()` it doesn't actually do the copy, so any in-place operations change the same underlying data.

If this was intended to be more immutable, maybe we can disallow in-place changes?

Reply:
> Yeah, when trying to do a proper copy of a TaggableCLArray with ary.copy() it doesn't actually do the copy, so any in-place operations change the same underlying data. However, the exact same code with ary.copy(queue=queue) works.

Aah I see, makes sense. Yep your approach looks correct to me. Thanks for the fix!

Reply:
> If this was intended to be more immutable, maybe we can disallow in-place changes?

I don't see a good reason to diverge from the underlying `cla.Array` behavior. IMO, we should continue allowing in-place updates.

Reply:
> Aah I see, makes sense. Yep your approach looks correct to me. Thanks for the fix!

Reworked this a bit 
* added a `wrap_cl_array` method that does the second half of the `copy` that just adds the tags
* Use `wrap_cl_array` instead of `copy` when an actual copy of the underlying data is not intended.
* `TaggableCLArray.copy` now calls the `super().copy` and then `wrap_cl_array` to add tags

It seems to be working as intended now. What do you think?